tests/ModelContextProtocol.Tests/Client/McpClientFactoryTests.cs

```diff
@@ -186,7 +188,67 @@ public async Task McpFactory_WithInvalidTransportOptions_ThrowsFormatException(s
         await Assert.ThrowsAsync<ArgumentException>(() => McpClientFactory.CreateAsync(config, _defaultOptions, cancellationToken: TestContext.Current.CancellationToken));
     }
 
-    private sealed class NopTransport : ITransport, IClientTransport
+    [Theory]
+    [InlineData(typeof(NopTransport))]
+    [InlineData(typeof(FailureTransport))]
+    public async Task CreateAsync_WithCapabilities(Type transportType)
```

Comment:
What is this test validating?

Reply:
This is for testing code around McpClientFactory with clientOptions (Capabilities and its handlers) as they are not covered by UTs @stephentoub

Reply:
As `ToChatClientArguments` supports text, image, audio content types, existing UTs doesn't fully cover them